### ISSUE TITLE
Additional database resource doesn't work using for each in Cloud SQL Postgres

### ISSUE THREAD

**Issue by Nofugnosis** [terraform-google-modules/terraform-google-sql-db#568]
### TL;DR

I ran terraform plan and the new database resource from additional database module didn't appear in the tf plan log.

### Expected behavior

A new database besides the default database should be create using the additional database module.

### Observed behavior

I don't see the database resource in the terraform plan log.

### Terraform Configuration

```hcl
# List of databases for multiple database creation 
  databases = { for db in var.additional_databases : db.name => db }

# Create Database In CloudSQL
resource "google_sql_database" "database" {
  depends_on = [google_project_service.apis, google_sql_database_instance.instance]
  project    = var.project_id
  name       = local.database_name
  instance   = google_sql_database_instance.instance.name
  charset    = var.database_charset
  collation  = var.database_collation
  count      = var.master_instance_name == "" ? 1 : 0
}

# Create additional databases
resource "google_sql_database" "additional_databases" {
  depends_on      = [google_project_service.apis, google_sql_database_instance.instance]
  for_each        = local.databases
  project         = var.project_id
  name            = each.value.name
  instance        = google_sql_database_instance.instance.name

}
```

### Terraform Version

```sh
Initializing modules...

Initializing the backend...

Initializing provider plugins...
- Using previously-installed hashicorp/random v3.6.0
- Using previously-installed hashicorp/null v3.2.2
- Using previously-installed hashicorp/google v4.84.0
- Using previously-installed hashicorp/google-beta v4.84.0

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, we recommend adding version constraints in a required_providers block
in your configuration, with the constraint strings suggested below.

* hashicorp/null: version = "~> 3.2.2"
* hashicorp/random: version = "~> 3.6.0"
```

### Additional information

Part of the terraform plan log result:

module.gcp_cloudsql.data.google_project.project: Refreshing state...
module.gcp_cloudsql.data.google_compute_network.vpc: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.gcp_cloudsql.google_project_service.apis[1] will be created
  + resource "google_project_service" "apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "sbx-226722-hep10-bc86864e"
      + service            = "sqladmin.googleapis.com"
    }

  # module.gcp_cloudsql.google_project_service.apis[2] will be created
  + resource "google_project_service" "apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "sbx-226722-hep10-bc86864e"
      + service            = "servicenetworking.googleapis.com"
    }

  # module.gcp_cloudsql.google_project_service.apis[3] will be created
  + resource "google_project_service" "apis" {
      + disable_on_destroy = false
      + id                 = (known after apply)
      + project            = "sbx-226722-hep10-bc86864e"
      + service            = "vpcaccess.googleapis.com"
    }

  # module.gcp_cloudsql.google_sql_database.database[0] will be created
  + resource "google_sql_database" "database" {
      + collation       = (known after apply)
      + deletion_policy = "DELETE"
      + id              = (known after apply)
      + instance        = "sbx-226722-hep10-bc86864e-dav7"
      + name            = "sbx-226722-hep10-bc86864e-davdb7"
      + project         = "sbx-226722-hep10-bc86864e"
      + self_link       = (known after apply)
    }

  # module.gcp_cloudsql.google_sql_database_instance.instance will be created
  + resource "google_sql_database_instance" "instance" {
      + available_maintenance_versions = (known after apply)
      + connection_name                = (known after apply)
      + database_version               = "POSTGRES_14"
      + deletion_protection            = false
      + dns_name                       = (known after apply)
      + encryption_key_name            = (known after apply)
      + first_ip_address               = (known after apply)
      + id                             = (known after apply)
      + instance_type                  = (known after apply)
      + ip_address                     = (known after apply)
      + maintenance_version            = (known after apply)
      + master_instance_name           = (known after apply)
      + name                           = name"
      + private_ip_address             = (known after apply)
      + project                        = "name"
      + psc_service_attachment_link    = (known after apply)
      + public_ip_address              = (known after apply)
      + region                         = "us-east1"
      + self_link                      = (known after apply)
      + server_ca_cert                 = (known after apply)
      + service_account_email_address  = (known after apply)

      + replica_configuration {
          + ca_certificate            = (known after apply)
          + client_certificate        = (known after apply)
          + client_key                = (known after apply)
          + connect_retry_interval    = (known after apply)
          + dump_file_path            = (known after apply)
          + failover_target           = (known after apply)
          + master_heartbeat_period   = (known after apply)
          + password                  = (sensitive value)
          + ssl_cipher                = (known after apply)
          + username                  = (known after apply)
          + verify_server_certificate = (known after apply)
        }

      + settings {
          + activation_policy     = "ALWAYS"
          + availability_type     = "REGIONAL"
          + connector_enforcement = (known after apply)
          + disk_autoresize       = true
          + disk_autoresize_limit = 0
          + disk_size             = 10
          + disk_type             = "PD_SSD"
          + pricing_plan          = "PER_USE"
          + tier                  = "db-g1-small"
          + version               = (known after apply)

          + backup_configuration {
              + binary_log_enabled             = false
              + enabled                        = true
              + point_in_time_recovery_enabled = false
              + start_time                     = (known after apply)
              + transaction_log_retention_days = 7

              + backup_retention_settings {
                  + retained_backups = 30
                  + retention_unit   = "COUNT"
                }
            }

          + database_flags {
              + name  = "cloudsql.iam_authentication"
              + value = "on"
            }
          + database_flags {
              + name  = "ssl_max_protocol_version"
              + value = "TLSv1.3"
            }
          + database_flags {
              + name  = "ssl_min_protocol_version"
              + value = "TLSv1.2"
            }

          + insights_config {
              + query_insights_enabled = false
              + query_plans_per_minute = (known after apply)
              + query_string_length    = 1024
            }

          + ip_configuration {
              + allocated_ip_range = "ip"
              + ipv4_enabled       = true
              + private_network    = "link"
              + require_ssl        = true
            }

          + location_preference {
              + zone = "us-east1-b"
            }

          + maintenance_window {
              + day  = 7
              + hour = 1
            }
        }

      + timeouts {
          + create = "60m"
          + delete = "60m"
          + update = "60m"
        }
    }

  # module.gcp_cloudsql.google_sql_ssl_cert.client_cert will be created
  + resource "google_sql_ssl_cert" "client_cert" {
      + cert               = (known after apply)
      + cert_serial_number = (known after apply)
      + common_name        = "client-name"
      + create_time        = (known after apply)
      + expiration_time    = (known after apply)
      + id                 = (known after apply)
      + instance           = "name"
      + private_key        = (sensitive value)
      + project            = "name"
      + server_ca_cert     = (known after apply)
      + sha1_fingerprint   = (known after apply)
    }

Plan: 28 to add, 0 to change, 0 to destroy.
